src/Vizkit3dWorld.cpp

```diff
@@ -36,12 +39,14 @@ Vizkit3dWorld::Vizkit3dWorld(std::string path,
     , zFar(zFar)
     , horizontalFov(horizontalFov)
 {
-    
     loadGazeboModelPaths(modelPaths);
+    if (!QApplication::instance()) {
+        LOG_WARN_S << "letting Vizkit3dWorld create the QApplication object" <<
+            " is deprecated. You should now create it before creating a Vizkit3dWorld" <<
+            std::endl;
 
-    int argc = 1;
-    char const*argv[] = { "vizkit3d_world" };
-    app = new QApplication(argc, const_cast<char**>(argv));
+        app = new QApplication(QT_ARGC, const_cast<char**>(QT_ARGV));
```

Comment:
Even if, technically, removing the `app` attribute would break backward compatibility, I think it would be best to remove the attribute, since after the first initialized vizkit3dworld object, it *will* be NULL. It would ensure that subclasses that use `app` will not compile anymore (instead of crashing when creating their second Vizkit3dWorld object).